src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -75,6 +77,11 @@ internal int CurrentPasteOffset
         private bool hasNodeInSyncWithDefinition;
         private Guid guid;
 
+        /// <summary>
+        /// Property is true after a workspace is loaded.
+        /// </summary>
+        private bool workspaceLoaded;
```

Comment:
perhaps we should rename this to computeUpstreamNodesWhenWorkspaceModified - or clarify in the comment that this is also set false when the graph is cleared.